
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Chapter X: Building a mechanistic model with the Antimony language &#8212; Biochemical Modeling</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Running Simulations" href="simulation.html" />
    <link rel="prev" title="Stoichiometric Networks" href="stoichiometricNetworks.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  <img src="_static/logo.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Biochemical Modeling</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Introduction to Biochemical Pathway Modeling
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="modeling.html">
   What is biochemical network modeling?
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="notebooks.html">
   Content with notebooks
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="stoichiometricNetworks.html">
   Stoichiometric Networks
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Chapter X: Building a mechanistic model with the Antimony language
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="simulation.html">
   Running Simulations
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="model_simulation.html">
   Chapter X: Running deterministic and stochastic simulations with Tellurium and libRoadRunner
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="plotting.html">
   Plotting Options
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="analysis.html">
   Analysis Tools
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="advanced_analyses.html">
   Chapter X: Advanced analyses with Tellurium
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="standardized_formats.html">
   Chapter X: SBML, SED-ML, and the COMBINE archive: standard formats supported in Tellurium
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/model_construction.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/model_construction.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-aggregation">
   Data aggregation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#modeling-biochemical-reaction-kinetics">
   Modeling biochemical reaction kinetics
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#writing-a-simple-antimony-string">
   Writing a simple Antimony string
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#editing-a-complex-model-of-a-true-biological-system">
   Editing a complex model of a true biological system
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="chapter-x-building-a-mechanistic-model-with-the-antimony-language">
<h1>Chapter X: Building a mechanistic model with the Antimony language<a class="headerlink" href="#chapter-x-building-a-mechanistic-model-with-the-antimony-language" title="Permalink to this headline">¶</a></h1>
<br>
<div align='center'>
    <figure>
        <img src="https://raw.githubusercontent.com/vporubsky/tellurium-libroadrunner-tutorial/master/data_aggregation_logo.png" width = "50%" style="padding: 0px">
        <figcaption>Model construction requires several stages including data aggregation, annotation and creation of a mathematical representation of the system which can be represented in a computational environment.</figcaption>
    </figure>
</div>
<br>
<br>
<p><span style="color:red">Include data aggregation, writing reactions, adding species and setting values</span>.</p>
<div class="section" id="data-aggregation">
<h2>Data aggregation<a class="headerlink" href="#data-aggregation" title="Permalink to this headline">¶</a></h2>
<p>Data aggregation is the collection of data from multiple experiments, scientific papers, and online data sources. Typically you will need to curate your aggregated data to ensure the quality of measurements are suffiencient to include in your model.</p>
<ul>
  <li>SABIO-RK: biochemical reaction kinetics database</li>
     <ul class="square">
      <li>Describes chemical reactions and kinetics</li>
      <li>Contains information about participants and modifiers in reactions</li>
      <li>Metabolic and signaling network reactions</li>
     </ul>
  <li>BRENDA: the comprehensive enzyme information system</li>
     <ul class="square">
      <li>Enzyme information classified by the biochemical reaction it catalyzes</li>
      <li>Kinetic information about substrates and products is available</li> 
     </ul>
  <li>ChEBI: dictionary of "small" chemical compounds</li>
  <li>KEGG: collection of pathway/genome/diesease/drug databases</li>
  <li>BioCYC: collection of pathway/genome databases</li>
       <ul class="square">
      <li>Search for genes, proteins, metabolites or pathways, and the occurence of your term will be located in multiple databases</li> 
     </ul>
  <li>BioModel: repository of mathematical models of biological systems</li>
      <ul class="square">
      <li> *Will be covered in more detail later in the course</li> 
     </ul>
</ul>
<p><a href="https://www.sciencedirect.com/science/article/abs/pii/S0958166917301428?via%3Dihub">Appendix A of Goldberg et al. (2018)</a> provides a useful and more comprehensive list of data sources containing intracellular biochemical data.</p>
<ul>
  <li>Metadata: data that describes biochemical data</li>
  <li>Collect information about:</li>
     <ul class="square">
      <li>Units</li>
      <li>Estimates of measurement accuracy</li>
      <li>Annotations</li>
      <li>Ontology terms defining the annotations</li>
      <li>etc.</li>
     </ul>
  <li>Collect provenance data:</li>
     <ul class="square">
      <li>Lab which generated the data</li>
      <li>Experimental conditions</li>
      <li>Protocol used to generate the data</li>
      <li>Paper which reported the measurement</li>
      <li>etc.</li><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">requests</span>
<span class="kn">import</span> <span class="nn">io</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">QUERY_URL</span> <span class="o">=</span> <span class="s1">&#39;http://sabiork.h-its.org/sabioRestWebServices/kineticlawsExportTsv&#39;</span>

<span class="c1"># Specify search fields and search terms</span>
<span class="n">query_dict</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;Organism&quot;</span><span class="p">:</span><span class="s1">&#39;&quot;SARS coronavirus&quot;&#39;</span><span class="p">,</span> <span class="s2">&quot;Product&quot;</span><span class="p">:</span><span class="s1">&#39;&quot;Spike glycoprotein RBD-SD1-ACE2 complex&quot;&#39;</span><span class="p">}</span>
<span class="n">query_string</span> <span class="o">=</span> <span class="s1">&#39; AND &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1">:</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">k</span><span class="p">,</span><span class="n">v</span><span class="p">)</span> <span class="k">for</span> <span class="n">k</span><span class="p">,</span><span class="n">v</span> <span class="ow">in</span> <span class="n">query_dict</span><span class="o">.</span><span class="n">items</span><span class="p">()])</span>

<span class="c1"># Specify output fields and send request</span>
<span class="n">query</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;fields[]&#39;</span><span class="p">:[</span><span class="s1">&#39;EntryID&#39;</span><span class="p">,</span> <span class="s1">&#39;Organism&#39;</span><span class="p">,</span> <span class="s1">&#39;UniprotID&#39;</span><span class="p">,</span><span class="s1">&#39;ECNumber&#39;</span><span class="p">,</span> <span class="s1">&#39;Parameter&#39;</span><span class="p">],</span> <span class="s1">&#39;q&#39;</span><span class="p">:</span><span class="n">query_string</span><span class="p">}</span>
<span class="n">request</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">post</span><span class="p">(</span><span class="n">QUERY_URL</span><span class="p">,</span> <span class="n">params</span> <span class="o">=</span> <span class="n">query</span><span class="p">)</span>
<span class="n">request</span><span class="o">.</span><span class="n">raise_for_status</span><span class="p">()</span>

<span class="c1"># Print query results</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">The results of SABIO-RK query for the formation of spike glycoprotein RBD-SD1-ACE2 complex in SARS coronavirus:</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">request</span><span class="o">.</span><span class="n">text</span><span class="p">)</span>
<span class="n">data_string</span> <span class="o">=</span> <span class="n">request</span><span class="o">.</span><span class="n">text</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">io</span><span class="o">.</span><span class="n">StringIO</span><span class="p">(</span><span class="n">data_string</span><span class="p">)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="n">parameter_types</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;parameter.type&#39;</span><span class="p">])</span>
<span class="k">if</span> <span class="s1">&#39;rate const.&#39;</span> <span class="ow">in</span> <span class="n">parameter_types</span><span class="p">:</span>
    <span class="n">k7</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;parameter.startValue&#39;</span><span class="p">])[</span><span class="n">parameter_types</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;rate const.&#39;</span><span class="p">)]</span>
    <span class="n">k7_units</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;parameter.unit&#39;</span><span class="p">])[</span><span class="n">parameter_types</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;rate const.&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">The formation of the spike glycoprotein RBD-SD1-ACE2 complex occurs at a rate of </span><span class="si">{</span><span class="n">k7</span><span class="si">:</span><span class="s2">.3f</span><span class="si">}</span><span class="s2"> </span><span class="si">{</span><span class="n">k7_units</span><span class="si">}</span><span class="s2">.&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The results of SABIO-RK query for the formation of spike glycoprotein RBD-SD1-ACE2 complex in SARS coronavirus:

EntryID	Organism	UniprotID	ECNumber	parameter.type	parameter.associatedSpecies	parameter.startValue	parameter.endValue	parameter.standardDeviation	parameter.unit
66773	SARS coronavirus	Q9BYF1 P59594	3.4.17.23	concentration	ACE2	3.13E-8	5.000000000000001E-7	-	M
66773	SARS coronavirus	Q9BYF1 P59594	3.4.17.23	Association constant	ACE2	362000.0		-	M^(-1)*s^(-1)
66773	SARS coronavirus	Q9BYF1 P59594	3.4.17.23	concentration	Spike glycoprotein RBD-SD1			-	-
66773	SARS coronavirus	Q9BYF1 P59594	3.4.17.23	Kd	ACE2	3.258E-7		-	M
66773	SARS coronavirus	Q9BYF1 P59594	3.4.17.23	rate const.		0.112		-	s^(-1)


The formation of the spike glycoprotein RBD-SD1-ACE2 complex occurs at a rate of 0.112 s^(-1).
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="modeling-biochemical-reaction-kinetics">
<h2>Modeling biochemical reaction kinetics<a class="headerlink" href="#modeling-biochemical-reaction-kinetics" title="Permalink to this headline">¶</a></h2>
<ul class="roman">
 <li>Speed of the biochemical reaction involved in the network determines evolution of the system</li>
  <li>Factors that influence the rates of reactions must be considered and modeled appropriately</li>
  <ul class="square">
  <li>e.g. the presence of an activator or repressor would change the baseline speed of that reaction</li>
 </ul>
 <li>Consider the mechanisms involved and determine appropriate simplifications</li>
</ul>
<ul class="roman">
 <li>Models which describe how variables in a system evolve over time</li>
 <ul class="square">
  <li>e.g. floating species concentrations</li>
 </ul>
 <li>Quantities are derived from the variables</li>
  <ul class="square">
  <li>e.g. pathway flux</li>
 </ul>
 <li>Some parameters of the model are fixed by the modeler</li>
   <ul class="square">
  <li>e.g. rate constants</li>
  <li>e.g. enzyme concentrations</li>
  <li>e.g. boundary species concentrations</li>
 </ul>
 <li>Deterministic differential equations models are useful when we can assume there are a large number of participants in the chemical reactions</li>
 <li>Stochastic models are useful for dilute systems in which reactions may not occur at every timepoint</li>
</ul>
<br>
<br>
<br>
<div align='center'><img src="https://raw.githubusercontent.com/vporubsky/tellurium-libroadrunner-tutorial/master/network_to_diff_eq.png" width = "60%" style="padding: 0px"></div></div>
<div class="section" id="writing-a-simple-antimony-string">
<h2>Writing a simple Antimony string<a class="headerlink" href="#writing-a-simple-antimony-string" title="Permalink to this headline">¶</a></h2>
<br>
Consider the system depicted below, in which species S1 is converted into species S2 with some chemical reaction rate, k1. Assuming this model follows mass-action kinetics as described in Chapter X, the rate law is shown: k1*[S1]. 
<br>
<div align='center'><img src="https://raw.githubusercontent.com/vporubsky/tellurium-libroadrunner-tutorial/master/simple-antimony.png" width="75%" style="padding: 20px"></div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">Ant_str</span> <span class="o">=</span> <span class="s2">&quot;&quot;&quot;</span>
<span class="s2">model test               # name the model</span>
<span class="s2">    compartment C1;      # specify compartments</span>
<span class="s2">    C1 = 1.0;            # assign compartment volume</span>
<span class="s2">    species S1, S2;      # specify species</span>
<span class="s2">    S1 in C1; S2 in C1;  # allocate species to appropriate compartment</span>

<span class="s2"> </span>
<span class="s2">    J1: S1 -&gt; S2; k1*S1; # reaction; reaction rate law;</span>
<span class="s2">    </span>
<span class="s2">    S1 = 10.0;           # assign species initial conditions</span>
<span class="s2">    S2 = 0.0;</span>

<span class="s2">    k1 = 1.0;            # assign constant values to global parameters</span>
<span class="s2">end</span>
<span class="s2">&quot;&quot;&quot;</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="editing-a-complex-model-of-a-true-biological-system">
<h2>Editing a complex model of a true biological system<a class="headerlink" href="#editing-a-complex-model-of-a-true-biological-system" title="Permalink to this headline">¶</a></h2>
<br>
It is common to build upon existing models to enable new analyses...
<br>
<div align='center'><img src="" width="75%" style="padding: 20px"></div><div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">antimony_str</span> <span class="o">=</span> <span class="s2">&quot;&quot;&quot;</span>
<span class="s2">model sars_cov2_infection</span>

<span class="s2">// Define model equations</span>
<span class="s2">production_genomic_ssRNA:               -&gt; gen_ssRNA;   k1*rep_ssRNA</span>
<span class="s2">transcription_genomic_ssRNA:  gen_ssRNA -&gt; rep_ssRNA;   k2*gen_ssRNA</span>
<span class="s2">translation_of_viral_envelope:          -&gt; envelope;    k3*rep_ssRNA</span>
<span class="s2">degradation_of_ssRNA:         rep_ssRNA -&gt; ;            k4*rep_ssRNA</span>
<span class="s2">degradation_of_envelope:       envelope -&gt; ;            k5*envelope</span>
<span class="s2">formation_of_free_virus_complex: gen_ssRNA + envelope -&gt; virus_free;  k6*gen_ssRNA*envelope</span>
<span class="s2">binding_of_virus_complex_to_host:  virus_free + $ACE2 -&gt; virus_bound; k7*virus_free*ACE2</span>

<span class="s2">// Annotate reactions</span>
<span class="s2">transcription_genomic_ssRNA.sboTerm = SBO:0000183</span>
<span class="s2">translation_of_viral_envelope.sboTerm = SBO:0000184</span>

<span class="s2">// Set global constants</span>
<span class="s2">k1 = 0.968; k2 = 0.025;</span>
<span class="s2">k3 = 1000; k4 = 0.222</span>
<span class="s2">k5 = 2; k6 = 7.5E-6</span>
<span class="s2">k7 = 0.112 </span>

<span class="s2">// Annotate kinetic constant distribution information</span>
<span class="s2"># k1.confidenceInterval = {0.883, 0.892}</span>
<span class="s2"># k4.confidenceInterval = {0.184, 0.186}</span>

<span class="s2">// Set initial conditions</span>
<span class="s2">rep_ssRNA = 1; gen_ssRNA = 0;</span>
<span class="s2">envelope = 0; ACE2 = 100;</span>
<span class="s2">virus_free = 0; virus_bound = 0;</span>

<span class="s2">end</span>
<span class="s2">&quot;&quot;&quot;</span>
</pre></div>
</div>
</div>
</div>
<br>
<html>
   <head>
      <title>Bibliography</title>
   </head>
   <body>
      <h1>Bibliography</h1>
      <ol>
         <li>
            <p>K. Choi et al., <cite>Tellurium: An extensible python-based modeling environment for systems and synthetic biology</cite>, Biosystems, vol. 171, pp. 74–79, Sep. 2018.</p>
         </li>
         <li>
            <p>E. T. Somogyi et al., <cite>libRoadRunner: a high performance SBML simulation and analysis library.,</cite>, Bioinformatics, vol. 31, no. 20, pp. 3315–21, Oct. 2015.</p>         
          <li>
            <p>L. P. Smith, F. T. Bergmann, D. Chandran, and H. M. Sauro, <cite>Antimony: a modular model definition language</cite>, Bioinformatics, vol. 25, no. 18, pp. 2452–2454, Sep. 2009.</p>
         </li>
         <li>
            <p>K. Choi, L. P. Smith, J. K. Medley, and H. M. Sauro, <cite>phraSED-ML: a paraphrased, human-readable adaptation of SED-ML</cite>, J. Bioinform. Comput. Biol., vol. 14, no. 06, Dec. 2016.</p>
         </li>
         <li>
            <p>J. Wolf, H. Y. Sohn, R. Heinrich, and H. Kuriyama, <cite>Mathematical analysis of a mechanism for autonomous metabolic oscillations in continuous culture of Saccharomyces cerevisiae</cite>, FEBS Lett., vol. 499, no. 3, pp. 230–234, Jun. 2001.</p>
         </li>
      </ol>
   </body>
</html></div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="stoichiometricNetworks.html" title="previous page">Stoichiometric Networks</a>
    <a class='right-next' id="next-link" href="simulation.html" title="next page">Running Simulations</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The Jupyter Book Community<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>